### PR TITLE
code-cleanup: move static string declaration scope

### PR DESCRIPTION
**Pull request by adisuissa**
Commit Message: code-cleanup: move static string declaration scope
Additional Description:
minor refactor to address internal fuzzers issues.

Risk Level: low
Testing: N/A
Docs Changes: N/A
Release Notes: N/A
Platform Specific Features: N/A

### CODE REVIEW

File: source/common/http/conn_manager_impl.cc [hunk 1]
```diff
@@ -792,6 +792,10 @@ absl::optional<uint64_t> ConnectionManagerImpl::HttpStreamIdProviderImpl::toInte
       *parent_.request_headers_);
 }
 
+namespace {
+static const std::string kRouteFactoryName = "envoy.route_config_update_requester.default";
```
Comment (yanavlasov) [inline source/common/http/conn_manager_impl.cc:796]:
```suggestion
constexpr absl::string_view kRouteFactoryName = "envoy.route_config_update_requester.default";
```

**Review by yanavlasov — CHANGES_REQUESTED** [review]
/wait

File: source/common/http/conn_manager_impl.cc (hunk 1 above)
Comment (adisuissa) [reply source/common/http/conn_manager_impl.cc]: Done, but kept `static`. LMK if you think it should not be.

**Review by yanavlasov — DISMISSED** [review]

**Comment by yanavlasov** [comment]
Code format is asking to change std::string_view to absl::string_view.

/wait

File: source/common/http/conn_manager_impl.cc (hunk 1 above)
Comment (yanavlasov) [reply source/common/http/conn_manager_impl.cc]: static is redundant in this context. It is not going to hurt but it is also not doing anything either. Anonymous namespace makes anything in it translation unit local.

Comment (adisuissa) [reply source/common/http/conn_manager_impl.cc]: Removed.

**Comment by ravenblackx** [comment]
@yanavlasov ping

**Comment by ravenblackx** [comment]
@yanavlasov another ping, this time also on slack.

**Review by yanavlasov — APPROVED** [review]

**Pull request merged**